examples/llm/src/models/mosaic_gpt/mosaic_gpt.py

```diff
@@ -364,6 +410,11 @@ def prepare_inputs_for_generation(self,
             raise NotImplementedError(
                 'MosaicGPT does not support generation with right padding.')
 
+        if self.attn_uses_sequence_id and self.training:
```

Comment:
would we ever be using the generate fn if model is in `self.training` mode?

Reply:
We might not end up doing so, but it seems like a user could. HF generate does not put the model into eval mode, as far as I can tell.